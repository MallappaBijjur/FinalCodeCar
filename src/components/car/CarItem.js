import React from 'react';
import '../home-car.css';

const CarItem = (props) => {
  const carInfo = `Stock #
              ${props.stockNumber} -
              ${props.mileage.number} ${props.mileage.unit.toUpperCase()} -
              ${props.fuelType} -
              ${props.color}`;
  const detaiPageLink = `/car/${props.stockNumber}`;
    return (<div className='car__item' data-testid={`car-${props.stockNumber}`}>
              <div className='car__item__img--container' >
                 <img
                  className='car__item--img'
                  src={props.pictureUrl} alt='Car'
                />
              </div>
              <div className='car__item__details--container'>
                <div className='car__item__details'>
                  <p className='car__item--info landing__top--text--bold margin__set'>{props.modelName}</p>
                  <div className='car__item__text car__item__text--small car__item__width--max '>
                    <p className="car__item__text">{carInfo}</p>
                  </div>
                  <a href={detaiPageLink} className="car__item__text car__item__text--link car__item__text--small pagelink__text--orange">View details</a>
                </div>
              </div>
            </div>)
}

export default CarItem;
