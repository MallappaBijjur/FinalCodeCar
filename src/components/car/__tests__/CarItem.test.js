import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, waitForElement } from '@testing-library/react';
import rerender from 'react-test-renderer';
import CarItem from '../CarItem';

describe("CarItem", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
    const carsData = {
					"stockNumber":30562,
					"manufacturerName":"Fiat",
					"modelName":"Ducato",
					"color":"yellow",
					"mileage":{
						 "number":100013,
						 "unit":"km"
					},
					"fuelType":"Petrol",
					"pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
			 };
    
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<CarItem  {...carsData} />, div);
	});
	
	it('renders Cars Details and Ducato as Model Name', async () => {
        const { getByText } = render(<CarItem {...carsData} />);   
		const linkElement = await waitForElement(() => getByText("Ducato"));
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<CarItem {...carsData} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
