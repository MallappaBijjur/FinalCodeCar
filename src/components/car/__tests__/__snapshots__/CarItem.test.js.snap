// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarItem matches snapshot 1`] = `
<div
  className="car__item"
  data-testid="car-30562"
>
  <div
    className="car__item__img--container"
  >
    <img
      alt="Car"
      className="car__item--img"
      src="https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
    />
  </div>
  <div
    className="car__item__details--container"
  >
    <div
      className="car__item__details"
    >
      <p
        className="car__item--info landing__top--text--bold margin__set"
      >
        Ducato
      </p>
      <div
        className="car__item__text car__item__text--small car__item__width--max "
      >
        <p
          className="car__item__text"
        >
          Stock #
              30562 -
              100013 KM -
              Petrol -
              yellow
        </p>
      </div>
      <a
        className="car__item__text car__item__text--link car__item__text--small pagelink__text--orange"
        href="/car/30562"
      >
        View details
      </a>
    </div>
  </div>
</div>
`;
